docs/reference/contributing/guidelines/guidelines.md

```diff
@@ -1,10 +1,45 @@
+
+### Licensing  
+
+A license is the contract between the author permitting the use of software to others. It specifies what you can and cannot do when receiving the software. It provides protection for both the user and owner of the software. In an Mbed project, the full terms of the license can be found in a file named LICENSE. Additionally, all source files must contain the SPDX identifier as a comment at the beginning of the file.  
+
+Note that one repository may contain multiple, independent codebases, each with their own license. If you are integrating two libraries with different licenses, it is important that each library retains its original license. In the case of a repository having software with multiple licenses, the contribution will be made according to the license of the file the contribution modifies. By creating a pull request on GitHub, you are agreeing to license your contributions under the same license as the original code. Is commonly reffered to as "inbound=outbound". This enables contributions to happen in a quick and effortless way and encourages collaboration.  
+
+Most Mbed OS software is licensed under a permissive license. The three most common permissive licenses are:  
+- Apache 2.0  
+- BSD 3-Clause  
+- MIT
+
+For new Mbed projects, we suggest adopting the Apache 2.0 license. Note that any Mbed software release under a permissive license cannot accept any code that is licensed under a "copyleft" license. Doing so would prevent us from continuing to distribute our code under the permissive license. You are welcome to use Mbed software with copyleft licenses, as long as the rules of that license are followed.  
+
+A more detailed description on licenses can be found in the [guidelines/contributing](../contributing/guidelines/license.html).
+
+### Tips  
+
+- The maintainers and reviewers are your friends. At times, programming can be very personal. However, it's important to realize that we all share a common goal, and that honest feedback is constructive feedback. 
+- Code consistency and maintainability is more important than functionality. The existing style of a codebase overrules any personal preference. 
+- Larger contributions take longer to be accepted than smaller contributions. The best contributions are small and purposeful, achieving a single goal. You may be asked to split up a contribution if it contains multiple unrelated changes. 
+- Consistency is an important aspect of a codebase. To ensure consistency in Mbed OS code, we have created contributing guidelines. Any contribution to Mbed OS needs to meet the following criteria:
```

Comment:
I wonder if line 37 should be moved into this list.

Reply:
will fix